File: solutions/c#/ChessGame/Rook.cs
namespace ChessGame
{
    public class Rook : Piece
    {
        public Rook(Color color, int row, int col) : base(color, row, col) { }

        public override bool CanMove(Board board, int destRow, int destCol)
        {
            return Row == destRow || Col == destCol;
        }
    }    
}

File: solutions/java/src/chessgame/pieces/Rook.java
package chessgame.pieces;

import chessgame.Board;
import chessgame.Cell;
import chessgame.Color;

public class Rook extends Piece {
    public Rook(Color color) {
        super(color);
    }

    @Override
    public boolean canMove(Board board, Cell from, Cell to) {
        return (from.getRow() == to.getRow() || from.getCol() == to.getCol());
    }
}